google/cloud/bigtable_v2/services/bigtable/async_client.py

```diff
@@ -1194,9 +1164,7 @@ def read_change_stream(
         # Certain fields should be provided within the metadata header;
         # add these here.
         metadata = tuple(metadata) + (
-            gapic_v1.routing_header.to_grpc_metadata(
-                (("table_name", request.table_name),)
-            ),
+            _cached_to_metadata((("table_name", request.table_name),)),
```

Comment:
Unless I'm mistaken in almost all cases metadata will start with being empty, why not include it in the cache key as well?

Reply:
My end goal was to get this fix upstreamed in api-core, so I was trying to keep the signature the same. 

I ended up [getting that through](https://github.com/googleapis/python-api-core/pull/526) though, so I removed this part from this PR